File: PersistenceAndServices/src/main/java/service_base/UserService.java
package service_base;

import de.mkammerer.argon2.Argon2;
import de.mkammerer.argon2.Argon2Factory;
import domain.User;
import repository.UserDatabaseRepository;
import service_base.validators.UserValidator;

public class UserService {
    private UserDatabaseRepository repository;
    private final UserValidator validator;

    public UserService(UserDatabaseRepository repository) {
        this.repository = repository;
        this.validator = new UserValidator();
    }

    public int login(User user) throws Exception {
        validator.validate(user);
        String hashedPassword = repository.getPasswordOfUser(user.getUsername());
        Argon2 A2 = Argon2Factory.create(16,32);
        if (A2.verify(hashedPassword,user.getPassword()))
            return 1;
        else throw new Exception("Invalid login!\n");
    }
}

File: MPPCS/Service/UserService.cs
using Isopoh.Cryptography.Argon2;
using CommonDomain;
using Persistence.Repository;
using Persistence.Service.Validator;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Persistence.Service
{
    public class UserService
    {
        private UserRepository repository;
        private UserValidator validator;

        public UserService(UserRepository repository)
        {
            this.repository = repository;
            this.validator = new UserValidator();
        }

        public int Login(User user)
        {
            validator.validate(user);
            string hashedPassword = repository.GetPasswordOfUser(user.GetUsername());            
            if (Argon2.Verify(hashedPassword, user.GetPassword()))
            {
                return 1;
            }
            else throw new Exception("Invalid login!\n");
        }
    }
}
